src/shared/bslSyntax.sml

```diff
@@ -196,8 +196,18 @@ struct
   val bprog = mk_BirProgram
     handle e => raise wrap_exn "bprog" e
   fun bprog_list obs_ty bl_list =
-    (curry2 mk_BirProgram_list)
-      obs_ty (List.map (uncurry3 ((curry4 mk_bir_block_list) obs_ty)) bl_list)
+    let
+      (* Instantiate the observation type for all statements *)
+      val bl_list_obs_ty = List.map
+        (fn (a, l_stmts, b) => (a, List.map (inst [alpha |-> obs_ty]) l_stmts, b))
```

Comment:
could it happen that the l_stmts already have some terms (statements) which don't have alpha as an observation type? for example when there is an observe statement?

Reply:
I haven't tried, so I should write tests, but if there is no alpha, then `inst` should have no effect.

Reply:
or alpha means something else possibly

Reply:
Oh, right. We should match on `'a bir_program_t` instead.

Reply:
yes, if `inst` is a pattern match, then this would make it stable

Reply:
are you going to implement the match on `'a bir_program_t`?

Reply:
@andreaslindner Did the new commit fix this?